components/core/src/clp/string_utils/string_utils.cpp

```diff
@@ -129,6 +129,27 @@ bool is_wildcard(char c) {
     return false;
 }
 
+auto replace_unescaped_char(
+        char const escape_char,
+        char const src_char,
+        char const target_char,
+        std::string& src_string
+) -> void {
+    bool escaped = false;
```

Comment:
nit. let's use braclet init.
```suggestion
    bool escaped{false};
```

Reply:
oh ok, why is that?

Reply:
The first one allows implict type conversion. For example, if you mistakenly write bool escape = -1; I believe it will work but it might not do what you intended it to do.

The second one require explicit type matching so you can't do something like bool escaped{-1}; (you can try and it should fail the compile).

There's also some implications on which constructor it calls. but for this case it is not that complicated. Try to use {} since it's our coding standard.